### PR TITLE
Support cursors for aggregation pipelines

### LINKED ISSUE #1892: Support cursors for aggregation pipelines
### What should be done?

We should add support for cursors for `aggregate` command the same way we do that for the `find` command. Let's hide it under [`if h.EnableCursors`](https://github.com/FerretDB/FerretDB/blob/d38ab72dcb0ff0a9cafd8c81050713211669b94a/internal/handlers/pg/msg_find.go#L136) too.

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #1892.

Also fixes `find` zero `batchSize` when it calls cursor.Next(). It should fetch all the rest of documents, not zero document.
## Readiness checklist

- [ ] I added/updated unit tests.
- [x] I added/updated integration/compatibility tests.
- [ ] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/pg/msg_find.go [hunk 1]
```diff
@@ -80,10 +80,6 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 
 	var resDocs []*types.Document
 	err = dbPool.InTransaction(ctx, func(tx pgx.Tx) error {
-		if params.BatchSize == 0 {
```
Comment (chilagrow) [inline internal/handlers/pg/msg_find.go:83]: See test `TestQueryBatchSize/ZeroBatchSize`

**Review by AlekSi — APPROVED** [review]
Code LGTM, but let's work on all comments

File: integration/query_test.go [hunk 2]
```diff
@@ -921,6 +921,25 @@ func TestQueryBatchSize(t *testing.T) {
 		require.Equal(t, 118, cursor.RemainingBatchLength())
 	})
 
+	t.Run("ZeroBatchSize", func(t *testing.T) {
+		t.Parallel()
+
+		// set BatchSize to 0
```
[the remark this answers is not in the document]
Comment (chilagrow) [reply integration/query_test.go]: Yeah just saw that. Valid point, removed comments and updated code in some places too

**Review by AlekSi — DISMISSED** [review]

**Review by noisersup — APPROVED** [review]
LGTM

File: integration/aggregate_documents_test.go [hunk 3]
```diff
@@ -659,3 +662,248 @@ func TestAggregateUnsetErrors(t *testing.T) {
...
+			require.Equal(t, tc.firstBatch, firstBatch)
+		})
+	}
+}
+
+func TestAggregateBatchSize(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup.Setup(t)
+
+	// the number of documents is set to much bigger than the default batchSize of 101
```
Comment (rumyantseva) [inline integration/aggregate_documents_test.go:824]: Strictly speaking, if we want to answer the question "why", in this comment, we should also mention that we want it to cause `getMore` needed to be called (if I understood the idea correctly).

**Review by rumyantseva — APPROVED** [review]
And the comment itself wasn't posted 😓

File: integration/aggregate_documents_test.go (hunk 3 above)
Comment (chilagrow) [reply integration/aggregate_documents_test.go]: Good point, nowhere currently mention about implicit call to `getMore`. Updated the test comments

**Review by AlekSi — APPROVED** [review]

**Review by noisersup — APPROVED** [review]
LGTM

**Review by rumyantseva — APPROVED** [review]
LGTM

**Pull request merged**